ui-core/src/main/java/com/adyen/checkout/components/ui/view/AdyenTextInputEditText.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 26/8/2019.
+ */
+package com.adyen.checkout.components.ui.view
+
+import android.content.Context
+import android.text.Editable
+import android.text.InputFilter
+import android.text.InputFilter.LengthFilter
+import android.text.TextWatcher
+import android.util.AttributeSet
+import androidx.annotation.CallSuper
+import com.google.android.material.R
+import com.google.android.material.textfield.TextInputEditText
+
+open class AdyenTextInputEditText @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
+    TextInputEditText(context, attrs, if (defStyleAttr == 0) R.attr.editTextStyle else defStyleAttr) {
+
+    private var mListener: Listener? = null
+
+    open val rawValue: String
+        get() = if (text != null) text.toString() else ""
+
+    /**
+     * Constructor of AdyenTextInputEditText.
+     */
+    init {
+        this.addTextChangedListener(textWatcher)
+    }
+
+    fun setOnChangeListener(listener: Listener) {
+        mListener = listener
+    }
+
+    @CallSuper
+    protected open fun afterTextChanged(editable: Editable) {
+        if (mListener != null) {
```

Comment:
`mListener?..onTextChanged(editable)` would be prettier

Oh and m can be dropped

Reply:
👍